### PR TITLE
feat(Toolbar): #749 Adding UI for ColumnPinning in the toolbar.

### PR DESCRIPTION
**Pull request by gedinakova**
Closes # 749.  

- Added UI for ColumnPinning to the toolbar.
- Extracted a ColumnChooserBase and ColumnChooserItemBase classes.

### CODE REVIEW

**Comment by ChronosSF** [comment]
I guess this is still in development because of the overlay integration?

**Comment by ChronosSF** [comment]
What do we need to do to have the overlay show correctly? Just enabling the toolbar in the pinning sample shows this:

pinning

**Comment by gedinakova** [comment]
@ChronosSF I updated the branch to not load the overlay modal and position it properly. However, there's an issue with the z-index, so currently the dropdown is displayed below the grid. This is expected to be fixed by PR#1695. Finally, the actual pinning through the toolbar UI will start working once the MultiColumnHeaders PR is merged along with the changes to the pinning functionality.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 1]
```diff
@@ -323,6 +323,9 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this._columnHiding = value;
```
Comment (rmkrmk) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: Please add a check whether the new value is different than the current value and apply the new value only if there is need.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 2]
```diff
@@ -399,6 +402,24 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     @Input()
     public columnHidingTitle = '';
 
+    @Input()
+    get columnPinning() {
+        return this._columnPinning;
+    }
+
+    set columnPinning(value) {
+        this._columnPinning = value;
```
Comment (rmkrmk) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: Please add a check whether the new value is different than the current value and apply the new value only if there is need.

**Pull request merged**